src/components/DeadOrAlive/DeadOrAlive.js

```diff
@@ -1,161 +1,132 @@
+        this.props.toggleScoreDisplay();
     }
 
     nextOne() {
         this.setState({
-            characterImg: loading,
-            characterName: 'loading...'
+            loading: true,
         }, () => {
-            this.getRandomCharacter();
-            this.toggleScoreDisplay();
+            this.setCharacter(() => {
+                this.props.toggleScoreDisplay();
+                this.toggleLoadingDisplay();
+            });
         })
     }
 
-    toggleScoreDisplay() {
+    toggleLoadingDisplay() {
         this.setState({
-            scoreDisplay: !this.state.scoreDisplay
-        })
+            loading: !this.state.loading
+        });
     }
 
-
     render() {
-
         return (
             <div>
-
                 {
-                    (this.state.scoreDisplay === false && this.state.currentTry < 11) ? (
-                        <DisplayGame
-                            currentTry={this.state.currentTry}
-                            characterImg={this.state.characterImg}
-                            characterName={this.state.characterName}
+                    (this.props.scoreDisplay === false && this.props.currentTry < 11 && this.state.loading === false) && (
+                        <Game
+                            currentTry={this.props.currentTry}
+                            character={this.state.character}
                             checkAnswer={this.checkAnswer}
                         />
-                    ) : false
+                    )
                 }
 
                 {
-                    (this.state.scoreDisplay && this.state.currentTry < 11) ? (
-                        <DisplayScore
-                            scoreStyle={this.state.scoreStyle}
-                            scoreMessage={this.state.scoreMessage}
-                            characterImg={this.state.characterImg}
-                            score={this.state.score}
+                    (this.props.scoreDisplay && this.props.currentTry < 11 && this.state.loading === false) && (
+                        <Score
+                            scoreStyle={this.props.scoreStyle}
+                            scoreMessage={this.props.scoreMessage}
+                            characterImg={this.state.character.image}
+                            score={this.props.score}
                             nextOne={this.nextOne}
                         />
-                    ) : false
+                    )
                 }
 
                 {
-                    (this.state.currentTry === 11) ? (
-                        <DisplayFinalScore
-                            scoreStyle={this.state.scoreStyle}
-                            scoreMessage={this.state.scoreMessage}
-                            characterImg={this.state.characterImg}
-                            score={this.state.score}
+                    (this.props.currentTry === 11 && this.state.loading === false) && (
+                        <FinalScore
```

Comment:
`<Score />` and `<FinalScore />` are very similar one to each other.
They have almost exact same API.
Maybe we could try to merge them together and control its presentation by means of inpt properties?

Reply:
I've merged them to one Score component